src/Executables/Game/World/Map.cs

```diff
@@ -279,22 +279,47 @@ private MonsterEntity SpawnMonster(uint id, SpawnPoint spawnPoint)
             var baseY = spawnPoint.Y;
             if (spawnPoint.RangeX != 0)
             {
-                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeX);
             }
 
             if (spawnPoint.RangeY != 0)
             {
-                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeY, spawnPoint.RangeY);
             }
 
-            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger, _itemManager,
+            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger,
+                _itemManager,
                 id,
-                (int) (PositionX + (baseX + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                (int) (PositionY + (baseY + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                RandomNumberGenerator.GetInt32(0, 360));
-            _world.SpawnEntity(monster);
+                0,
+                0
+            );
+
+            if (((uint)EAiFlags.NoMove & monster.Proto.AiFlag) == (uint)EAiFlags.NoMove) // no_move
+            {
+                monster.PositionX = (int)(PositionX + baseX * SPAWN_POSITION_MULTIPLIER);
+                monster.PositionY = (int)(PositionY + baseY * SPAWN_POSITION_MULTIPLIER);
+                // SpawnPoint.Direction follows the compass system with counter-clockwise increments: 1 = South, 2 = South-East, 3 = East, 5 = North, etc.
+                var compassDirection = spawnPoint.Direction - 1;
+                var rotation = 45 * (compassDirection < 0 ? 8 : compassDirection);
+                monster.Rotation = rotation;
+            }
+            else
+            {
+                monster.PositionX = (int) PositionX + (baseX + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) * SPAWN_POSITION_MULTIPLIER;
+                monster.PositionY = (int) PositionY + (baseY + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) * SPAWN_POSITION_MULTIPLIER;
+            }
+
+            if (monster.Rotation == 0)
+            {
+                monster.Rotation = RandomNumberGenerator.GetInt32(0, 360);
+            }
+            
```

Comment:
I know the code above is a bit of a mess. I'm feeling an irresistible urge to refactor it. :)

Reply:
It's good enough for now :)